pkg/registry/softwarecomposition/collapseconfiguration/strategy_test.go

```
@@ -0,0 +1,148 @@
+	// Pass a different type to confirm the type assertion fails cleanly.
+	notACC := &softwarecomposition.ApplicationProfile{}
+	errs := s.Validate(context.Background(), notACC)
+	if len(errs) != 1 {
+		t.Fatalf("expected 1 internal error for type mismatch, got: %v", errs)
+	}
+}
+
+func TestValidateUpdate_SameRules(t *testing.T) {
+	s := NewStrategy(newScheme())
+	old := &softwarecomposition.CollapseConfiguration{Spec: softwarecomposition.CollapseConfigurationSpec{OpenDynamicThreshold: 50}}
+	new := &softwarecomposition.CollapseConfiguration{
```

golangci-lint (2.11.4)

[error] variable new has same name as predeclared identifier

(predeclared)